File: src/main/java/ru/chernov/urlshortener/UrlShortenerApplication.java
package ru.chernov.urlshortener;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import ru.chernov.urlshortener.config.properties.LinkProperties;
import ru.chernov.urlshortener.config.properties.RedisProperties;


@EnableConfigurationProperties({
        LinkProperties.class,
        RedisProperties.class
})
@EnableJpaRepositories
@SpringBootApplication
public class UrlShortenerApplication {

    public static void main(String[] args) {
        SpringApplication.run(UrlShortenerApplication.class, args);
    }

}

File: src/main/kotlin/ru/chernov/urlshortener/UrlShortenerApplication.kt
package ru.chernov.urlshortener

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication
import ru.chernov.urlshortener.config.properties.LinkProperties
import ru.chernov.urlshortener.config.properties.RedisProperties

@EnableConfigurationProperties(LinkProperties::class, RedisProperties::class)
@SpringBootApplication
class UrlShortenerApplication

fun main() {
    runApplication<UrlShortenerApplication>()
}